src/contract.rs

```diff
@@ -394,6 +434,51 @@ fn get_cw20_increase_allowance_msg(
     Ok(exec_allowance.into())
 }
 
+pub fn execute_update_config(
+    deps: DepsMut,
+    info: MessageInfo,
+    new_owner: Option<String>,
+    lp_fee_percent: Decimal,
+    protocol_fee_percent: Decimal,
+    protocol_fee_recipient: String,
+) -> Result<Response, ContractError> {
+    let owner = OWNER.load(deps.storage)?;
+    if Some(info.sender) != owner {
+        return Err(ContractError::Unauthorized {});
+    }
+
+    let new_owner_addr = new_owner
+        .as_ref()
+        .map(|h| deps.api.addr_validate(h))
+        .transpose()?;
+    OWNER.save(deps.storage, &new_owner_addr)?;
```

Comment:
Saving Owner configuration should happen after all the validation checks have passed

Reply:
What's the advantage in doing this? If validation fails later then the state change would be reverted anyways